### PR TITLE
(PCP-774) Add tasks-cache to puppet-specifications

### PR DESCRIPTION
**Pull request by ekinanp**

### CODE REVIEW

File: file_paths.md [hunk 1]
```diff
@@ -146,6 +146,7 @@ The files annotated by an '*' indicate that they are created by package installa
         modules *
             pxp-module-puppet *
         spool *                            # directory containing results of pxp-agent modules
+        tasks-cache *                      # directory containing a cache of all the files required for a specific task
```
Comment (MikaelSmith) [inline file_paths.md]: Not accurate. It contains a cache of files for all downloaded tasks.

File: file_paths.md [hunk 2]
```diff
@@ -252,6 +253,7 @@ create a `puppet` user or group.
                 pxp-agent.log                 # enabled by default
                 pxp-access.log                # not enabled by default
             spool *                           # directory containing results of pxp-agent modules
+            tasks-cache *                     # directory containing a cache of files for all downloaded tasks
```
Comment (ekinanp) [inline file_paths.md]: @MikaelSmith Is this what you meant?

Comment (MikaelSmith) [inline file_paths.md]: Yes. On Windows, the tasks-cache wasn't in the var directory, it was at pxp-agent/tasks-cache.

File: file_paths.md [hunk 3]
```diff
@@ -253,6 +254,7 @@ create a `puppet` user or group.
                 pxp-access.log                # not enabled by default
             spool *                           # directory containing results of pxp-agent modules
             run *
+        tasks-cache *                         # directory containing a cache of files for all downloaded tasks
```
Comment (ekinanp) [inline file_paths.md]: @MikaelSmith

Comment (Iristyle) [inline file_paths.md]: I'm wondering if this should be a sibling directory to `spool` under `var`, so that's closer to the *nix layout?

Comment (Iristyle) [inline file_paths.md]: I realize @MikaelSmith asked you to change this in a prior comment... but I'm wondering what went into that decision?

Comment (MikaelSmith) [inline file_paths.md]: Putting the `spool` dir under `var` on Windows looked like a mistake to me. On *nix it'd be under `opt`. However, we could choose to make it consistent even though it's kind of weird.

Comment (MikaelSmith) [inline file_paths.md]: Should I propose moving the spool dir as well? Any objection to moving that direction, anyway?

**Pull request merged**